src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -5594,3 +5600,47 @@ ExtractParametersFromParamList(ParamListInfo paramListInfo,
 																   parameterData->value);
 	}
 }
+
+
+/*
+ * Delegated functions which are not part of another transaction are naturally safe,
+ * but if it's part of an outer transaction (a 2PC), we should ensure that the current
+ * task will not do remote(connection) activity to ensure the transactional integrity
+ * of the current participating 2PC.
+ */
+static void
+EnsureDelegatedFunctionSafeExecution(List *taskList)
```

Comment:
This function seems not specific to delegated function calls. Maybe something like: EnsureNoRemoteExecutionFromWorkers

Reply:
Done